sources/api/schnauzer/src/helpers.rs

```diff
@@ -422,6 +400,163 @@ pub fn default(
     Ok(())
 }
 
+/// Gets the value at `idx` and unwraps it. Assumes you have already checked that `idx` is in range
+/// and says so in an error message if the value cannot be obtained.
+fn get_param<'a>(helper: &'a Helper<'_, '_>, idx: usize) -> Result<&'a Value, RenderError> {
+    Ok(helper
+        .param(idx)
+        .map(|v| v.value())
+        .context(error::Internal {
+            msg: "Missing param after confirming there are enough",
```

Comment:
```suggestion
        .context(error::MissingParam {
            number: idx,
            helper_name: helper.name(),
```
We can't assume they've checked for enough parameters anymore, so I think this makes the failure clearer.

(Also, should the get_param/template_name refactor be a separate, earlier commit?)